### PR TITLE
Move getSingleElementDeclFieldType from TFPartition to TFUtilties.

### PR DESCRIPTION
**Pull request by bgogul**

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/TFUtilities.h [hunk 1]
```diff
@@ -275,6 +275,12 @@ struct GraphGlobalConfiguration {
     return convertSwiftTypeToTF(ty) != 0;
   }
 
+  /// Given a decl for a struct that has a single field (typically because it is
+  /// known to be a standard library type like Int or Float), return the
+  /// canonical type of the single member, asserting and aborting if we get
+  /// something unexpected.
+  CanType getSingleElementDeclFieldType(NominalTypeDecl *decl);
```
Comment (slavapestov) [inline lib/SILOptimizer/Mandatory/TFUtilities.h]: If this really does have to be a struct, why not declare it as StructDecl?

**Review by mhong — APPROVED** [review]
I assume new call-site(s) will appear in a later PR?

**Review by lattner — APPROVED** [review]

File: lib/SILOptimizer/Mandatory/TFUtilities.h (hunk 1 above)
Comment (bgogul) [reply lib/SILOptimizer/Mandatory/TFUtilities.h]: It turns out that I did not need this utility after all. So, I am abandoning this PR.  

In any case, to answer your question, the existing use sites only have a pointer to NominalTypeDecl and I did not want to complicate this PR as TFUtilities will be cleaned up once we reach some stable state anyway.

**Pull request closed without merging**